flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/declarative/DeclarativeSchedulerNG.java

```diff
@@ -114,19 +115,22 @@
 import java.util.Collection;
 import java.util.HashSet;
 import java.util.Iterator;
+import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionException;
 import java.util.concurrent.Executor;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 import java.util.function.Function;
+import java.util.stream.StreamSupport;
 
 /** Declarative scheduler ng. */
 public class DeclarativeSchedulerNG implements SchedulerNG {
 
     private static final Logger LOG = LoggerFactory.getLogger(DeclarativeSchedulerNG.class);
 
+    // note: do not mutate this jobGraph!
```

Comment:
A less error-prone alternative might be to compute a constant `JobGraphJobInformation` from the initial JobGraph. We need few enough bits of information that this shouldn't be a problem.